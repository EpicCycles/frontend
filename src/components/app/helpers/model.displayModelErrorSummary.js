import {customerAddressFields} from "../model/helpers/fields";
import {displayModelErrorSummary} from "../model/helpers/model";

test('if no error or error_detail undefined returned', () => {
    const model = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6
    };
    const result = displayModelErrorSummary(model, customerAddressFields);
    expect(result).toBe(undefined);
});
test('if just error no error_detail error returned', () => {
    const model = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6,
        error: "Model level error"
    };
    const result = displayModelErrorSummary(model, customerAddressFields);
    expect(result).toBe("Model level error");
});
test('if no error with error_detail error_details formatted returned', () => {
    const model = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6,
        error_detail: {
            "address1": ["A valid address is required.", "SHow me the address is required."],
            "postcode": ["This field may not be blank."]
        },
    };
    const expectedText = "Address: A valid address is required. SHow me the address is required.<br>Postcode: This field may not be blank.";
    const result = displayModelErrorSummary(model, customerAddressFields);
    expect(result).toBe(expectedText);
});
test('if error and error_detail formatted returned', () => {
    const model = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6,
        error: "Model level error",
        error_detail: {
            "address1": ["A valid address is required.", "SHow me the address is required."],
            "postcode": ["This field may not be blank."]
        },
    };
    const expectedText = "Model level error<br>Address: A valid address is required. SHow me the address is required.<br>Postcode: This field may not be blank.";
    const result = displayModelErrorSummary(model, customerAddressFields);
    expect(result).toBe(expectedText);
});