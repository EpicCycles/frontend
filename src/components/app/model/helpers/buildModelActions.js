/**
 * buildModelActions - builds actions for the current model (use to build additional actions)
 * @param requiredActions - array of objects {iconName, iconTitle, requiresIdOnly, actionFunction}
 * @param model - model being displayed oredited
 * @returns array of model actions specific to current model
 */
export const buildModelActions = (requiredActions, model) => {
  return requiredActions.map(action => {
    const finalAction = {
      iconName: action.iconName,
      iconTitle: action.iconTitle,
    };
    const isSavedModel = model && model.id;
    if (action.requiresIdOnly) {
      finalAction.iconAction = () => isSavedModel && action.actionFunction(model.id);
    } else {
      finalAction.iconAction = () => isSavedModel && action.actionFunction(model);
    }
    finalAction.iconDisabled = !isSavedModel;
    return finalAction;
  });
};
