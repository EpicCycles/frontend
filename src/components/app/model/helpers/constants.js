export const POSTCODE_RULES = [
  { countryCode: 'AT', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'AU', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'BE', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'BG', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'CA', regex: /^[A-Za-z][0-9][A-Za-z] [0-9][A-Za-z][0-9]/, display: 'ANA NAN' },
  { countryCode: 'CH', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'CY', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'CZ', regex: /^\d{3}\s\d{2}$/, display: 'NNN NN' },
  { countryCode: 'DE', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'DK', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'EE', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'ES', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'FI', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'FR', regex: /^\d{5}$/, display: 'NNNNN' },
  {
    countryCode: 'GB',
    regex: /^[A-Z](([0-9][0-9]?)|([A-Z][0-9][0-9]?)|([A-Z]?[0-9][A-Z])) ?[0-9][ABD-HJLNP-UW-Z]{2}$/,
    display: 'CCNN NCC',
  },
  { countryCode: 'GR', regex: /^\d{3}\s\d{2}$/, display: 'NNNNN or NNN NN' },
  { countryCode: 'HU', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'IT', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'KR', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'LT', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'LU', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'MT', regex: /^\[A-Z]{3}\s\d{4}$/, display: 'AAA NNNN' },
  { countryCode: 'MY', regex: /^\d{5}$/, display: 'NNNNN' },
  { countryCode: 'NO', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'NZ', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'PH', regex: /^\d{4}$/, display: 'NNNN' },
  { countryCode: 'PL', regex: /^\d{2}-\d{3}$/, display: 'NN-NNN' },
  { countryCode: 'PT', regex: /^\d{4}-\d{3}$/, display: 'NNNN-NNN' },
  { countryCode: 'RO', regex: /^\d{6}$/, display: 'NNNNNN' },
  { countryCode: 'SE', regex: /^\d{3,4}$/, display: 'NNNNN or NNNN' },
  { countryCode: 'SG', regex: /^\d{6}$/, display: 'NNNNNN' },
  { countryCode: 'SI', regex: /^\d{3}\s\d{2}$/, display: 'NNN NN' },
  { countryCode: 'SK', regex: /^\d{3}\s\d{2}$/, display: 'NNN NN' },
  { countryCode: 'US', regex: /^(\d{5}([-]\d{4})?)$/, display: 'NNNNN-NNNN or NNNNN' },
  { countryCode: 'ZA', regex: /^\d{4}$/, display: 'NNNN' },
];
