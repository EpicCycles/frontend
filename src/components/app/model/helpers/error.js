export const BRAND_NAME_MISSING = "A name is required for the brand";
export const FRAME_NAME_MISSING = "A name is required for the frame";
export const ATTRIBUTE_NAME_MISSING = "A name is required for the attribute";
export const BUNDLE_NAME_MISSING = "A name is required for the bundle";
export const MODEL_NAME_MISSING = "A name is required for the model";
export const PART_NAME_MISSING = "A name is required for the part";
export const PART_TYPE_NAME_MISSING = "A name is required for the part type";
export const SECTION_MISSING = "A section must be selected for the part type";
export const PART_TYPE_MISSING = "A part type must be selected";
export const BRAND_MISSING = "A brand must be selected";
export const PRODUCTS_MISSING = "At least 2 products must be selected";
export const SUPPLIER_MISSING = "A supplier must be selected";
export const PART_MISSING = "A part must be selected";
export const ADDRESS_MISSING = "At least one line of address must be entered";
export const COUNTRY_MISSING = "A country must be selected";
export const INVALID_POSTCODE = "A postcode is expected which for this country has the format ";
export const SELECT_ONE_MISSING = "Please select an option.";
export const VALUE_MISSING = "A value is required.";
export const INVALID_EMAIL = "Invalid email address.";
export const INVALID_URL = "Invalid URL";
