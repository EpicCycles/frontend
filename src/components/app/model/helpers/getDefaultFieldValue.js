export const getDefaultFieldValue = modelField => {
  if (modelField.selectList && Array.isArray(modelField.selectList)) {
    const defaultValue = modelField.selectList.filter(listitem => listitem.isDefault);
    if (defaultValue.length > 0) return defaultValue[0].value;
  }
  return modelField.default;
};
