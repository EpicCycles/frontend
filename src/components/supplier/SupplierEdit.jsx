import React, { Fragment } from 'react';
import { Icon } from 'semantic-ui-react';
import { updateObject } from '../../helpers/utils';
import { checkForChangesAllFields, getModelKey, updateModel } from '../app/model/helpers/model';
import { supplierFields } from '../app/model/helpers/fields';
import EditModelPage from '../app/model/EditModelPage';
import ModelEditIcons from '../app/model/ModelEditIcons';
import * as PropTypes from 'prop-types';
import {getBrandsForSupplier} from "./helpers/supplier";

class SupplierEdit extends React.Component {
  state = {
    supplier: updateObject(this.props.supplier),
    persistedSupplier: this.props.supplier,
  };

  static getDerivedStateFromProps(props, state) {
    // Any time the current user changes,
    // Reset any parts of state that are tied to that user.
    // In this simple example, that's just the email.
    if (checkForChangesAllFields(supplierFields, props.supplier, state.persistedSupplier)) {
      return {
        supplier: updateObject(props.supplier),
        persistedSupplier: props.supplier,
      };
    }
    return null;
  }

  handleInputChange = (fieldName, input) => {
    let { supplier } = this.state;
    supplier = updateModel(supplier, supplierFields, fieldName, input);
    this.setState({ supplier });
  };

  onClickReset = () => {
    const { persistedSupplier } = this.state;
    const supplier = updateObject(persistedSupplier);
    this.setState({ supplier });
  };

  saveOrCreateSupplier = () => {
    let { supplier } = this.state;
    this.props.saveSupplier(supplier);
    if (this.props.closeModal) {
      this.props.closeModal();
    }
  };
  deleteOrRemoveSupplier = () => {
    let { supplier } = this.state;
    if (supplier && supplier.id) {
      this.props.deleteSupplier(supplier.id);
    }
    this.setState({ supplier: {}, persistedSupplier: {} });
    if (this.props.closeModal) {
      this.props.closeModal();
    }
  };

  render() {
    const { supplier, persistedSupplier } = this.state;
    const { closeModal, brands } = this.props;
    const componentKey = getModelKey(supplier);
    const brandNames = getBrandsForSupplier(componentKey, brands);
    return (
      <Fragment>
        {closeModal && (
          <div style={{ width: '100%', textAlign: 'right' }}>
            <Icon name="remove" circular link onClick={closeModal} />
          </div>
        )}
        <div style={{ width: '100%', textAlign: 'left' }}>
          <h2>Edit Supplier</h2>
          <EditModelPage
            model={supplier}
            persistedModel={persistedSupplier}
            modelFields={supplierFields}
            onChange={this.handleInputChange}
            showReadOnlyFields
          />
          <div key={`supplierBrands${componentKey}`}>Brands: {brandNames.join(', ')}</div>
          <div className="align_right">
            <ModelEditIcons
              componentKey={componentKey}
              model={supplier}
              modelSave={this.saveOrCreateSupplier}
              modelDelete={this.deleteOrRemoveSupplier}
              modelReset={this.onClickReset}
            />
          </div>
        </div>
      </Fragment>
    );
  }
}
SupplierEdit.defaultProps = {
  brands: [],
  supplier: {},
};
SupplierEdit.propTypes = {
  brands: PropTypes.array,
  supplier: PropTypes.object,
  componentKey: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  deleteSupplier: PropTypes.func,
  saveSupplier: PropTypes.func,
  closeModal: PropTypes.func,
};
export default SupplierEdit;
