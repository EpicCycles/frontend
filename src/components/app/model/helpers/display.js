import {
  BIKE,
  BRAND,
  CHECKBOX,
  CURRENCY,
  CUSTOMER,
  DATE_TIME,
  NUMBER,
  PART_TYPE,
  RADIO,
  SELECT_ONE,
  SUPPLIER,
  USER,
} from './fields';
import { getCustomerName } from '../../../customer/helpers/customer';
import { getBrandName } from '../../../brand/helpers/brand';
import { getNameForValue } from './model';
import { getSupplierName } from '../../../supplier/helpers/supplier';
import { getPartTypeName } from '../../../partType/helpers/partType';
import { getBikeName } from '../../../bike/helpers/bike';
import { getUserName } from '../../../user/helpers/user';
import { CHARGE } from '../../../quoteCharge/helpers/quoteChargeFields';
import { chargeName } from '../../../charge/helpers/chargeName';
import { findObjectWithKey } from '../../../../helpers/utils';
import { fittingText } from '../../../fitting/helpers/fitting';
import { FITTING } from '../../../quote/helpers/quoteFields';

export const fixedHeaderClassname = lockColumn => {
  if (lockColumn) return 'grid-header--fixed-left';
  return '';
};
export const fixedDetailsClassname = lockColumn => {
  if (lockColumn) return 'grid-item--fixed-left';
  return '';
};
export const gridHeaderClass = (baseClassName = '', fieldIndex, firstColumnLocked) => {
  const shouldLock = firstColumnLocked && fieldIndex === 0;
  return `${baseClassName} grid-item--header ${fixedHeaderClassname(shouldLock)}`;
};
export const gridItemClass = (baseClassName = '', fieldIndex, firstColumnLocked) => {
  const shouldLock = firstColumnLocked && fieldIndex === 0;
  return `grid-item ${fixedDetailsClassname(shouldLock)} ${baseClassName}`;
};
export const formattedDate = date => {
  if (date) return date.toLocaleDateString('en-GB');
  return '';
};
export const formattedDateTime = date => {
  if (date) return date.toLocaleString('en-GB');
  return '';
};
export const formattedPrice = fieldValue => {
  if (fieldValue)
    return Number(fieldValue).toLocaleString('en-GB', {
      style: 'currency',
      currency: 'GBP',
    });
  return '';
};
export const fieldAlignment = field => {
  switch (field.type) {
    case NUMBER:
    case CURRENCY:
      return 'align_right';
    case CHECKBOX:
    case RADIO:
      return 'align_center';
    default:
      return '';
  }
};
export const buildViewString = (model, field, sourceDataArrays = {}) => {
  let viewData;
  const {
    sections,
    brands,
    suppliers,
    customers,
    bikes,
    frames,
    users,
    charges,
    fittings,
  } = sourceDataArrays;
  const fieldValue = model ? model[field.fieldName] : undefined;
  switch (field.type) {
    case CUSTOMER:
      viewData = fieldValue ? getCustomerName(fieldValue, customers) : '';
      break;
    case CURRENCY:
      viewData = formattedPrice(fieldValue);
      break;
    case CHECKBOX:
      viewData = fieldValue ? 'Yes' : 'No';
      break;
    case DATE_TIME:
      viewData = fieldValue ? formattedDateTime(new Date(fieldValue)) : '';
      break;
    case PART_TYPE:
      viewData = fieldValue ? getPartTypeName(fieldValue, sections) : '';
      break;
    case BRAND:
      viewData = fieldValue ? getBrandName(fieldValue, brands) : '';
      break;
    case SUPPLIER:
      viewData = fieldValue ? getSupplierName(fieldValue, suppliers) : '';
      break;
    case BIKE:
      viewData = fieldValue ? getBikeName(fieldValue, bikes, frames, brands) : '';
      break;
    case SELECT_ONE:
      viewData = fieldValue ? getNameForValue(fieldValue, field.selectList) : '';
      break;
    case USER:
      viewData = fieldValue ? getUserName(fieldValue, users) : '';
      break;
    case CHARGE:
      viewData = fieldValue ? chargeName(fieldValue, charges) : '';
      break;
    case FITTING:
      const fitting = fieldValue ? findObjectWithKey(fittings, fieldValue) : undefined;
      viewData = fitting ? fittingText(fitting) : '';
      break;
    default:
      viewData = fieldValue ? fieldValue : '';
  }
  return viewData;
};
