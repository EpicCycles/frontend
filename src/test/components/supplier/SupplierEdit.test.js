import React from 'react';
import toJson from 'enzyme-to-json';
import SupplierEdit from '../../../components/supplier/SupplierEdit';
import {VALUE_MISSING} from "../../../components/app/model/helpers/error";

describe('SupplierEdit', () => {
  it('displays correctly for new supplier', () => {
    const component = shallow(<SupplierEdit />);
    expect(toJson(component)).toMatchSnapshot();
  });
  it('displays correctly for existing supplier', () => {
    const supplier = {
      id: 1,
      supplier_name: 'supplier name',
      brand_names: ['brands 1', 'brand 2', 'brand 3'],
      link: 'supplier.co.uk',
      preferred_supplier: false,
    };
    const component = shallow(<SupplierEdit supplier={supplier} />);
    expect(toJson(component)).toMatchSnapshot();
  });
  it('displays correctly for existing supplier as a modal dialogue', () => {
    const supplier = {
      id: 1,
      supplier_name: 'supplier name',
      brand_names: ['brands 1', 'brand 2', 'brand 3'],
      link: 'supplier.co.uk',
      preferred_supplier: false,
    };
    const closeModal = jest.fn();
    const component = shallow(<SupplierEdit supplier={supplier} closeModal={closeModal} />);
    expect(toJson(component)).toMatchSnapshot();
  });
  it('removing name for a supplier leads to an error being shown', () => {
    const supplier = {
      id: 1,
      supplier_name: 'supplier name',
      brand_names: ['brands 1', 'brand 2', 'brand 3'],
      link: 'supplier.co.uk',
      preferred_supplier: false,
    };
    const component = shallow(<SupplierEdit supplier={supplier} />);
    const supplierUpdated = {
      id: 1,
      supplier_name: '',
      brand_names: ['brands 1', 'brand 2', 'brand 3'],
      link: 'supplier.co.uk',
      preferred_supplier: false,
      error_detail: { supplier_name: VALUE_MISSING },
      changed: true,
    };
    component.instance().handleInputChange('supplier_name', '');
    expect(component.state('supplier')).toEqual(supplierUpdated);

    component.instance().onClickReset();
    expect(component.state('supplier')).toEqual(supplier);
  });
  it('save for existing supplier saves all changes', () => {
    const saveSupplier = jest.fn();
    const closeModal = jest.fn();
    const supplier = {
      id: 1,
      supplier_name: 'supplier name',
      brand_names: ['brands 1', 'brand 2', 'brand 3'],
      link: 'supplier.co.uk',
      preferred_supplier: false,
    };
    const component = shallow(
      <SupplierEdit supplier={supplier} saveSupplier={saveSupplier} closeModal={closeModal} />,
    );

    component.instance().handleInputChange('supplier_name', 'Replacement name');
    component.instance().handleInputChange('preferred_supplier', true);
    component.instance().handleInputChange('link', 'link.supplier.co.uk');
    component.instance().saveOrCreateSupplier();
    expect(saveSupplier.mock.calls).toHaveLength(1);
    expect(closeModal.mock.calls).toHaveLength(1);
    expect(saveSupplier.mock.calls[0][0].id).toBe(supplier.id);
    expect(saveSupplier.mock.calls[0][0].supplier_name).toBe('Replacement name');
    expect(saveSupplier.mock.calls[0][0].brand_names).toBe(supplier.brand_names);
    expect(saveSupplier.mock.calls[0][0].link).toBe('link.supplier.co.uk');
    expect(saveSupplier.mock.calls[0][0].preferred_supplier).toBe(true);
  });
  it('save for new supplier saves all changes', () => {
    const saveSupplier = jest.fn();
    const closeModal = jest.fn();
    const supplier = {};
    const component = shallow(
      <SupplierEdit supplier={supplier} saveSupplier={saveSupplier} closeModal={closeModal} />,
    );

    component.instance().handleInputChange('supplier_name', 'Replacement name');
    component.instance().handleInputChange('preferred_supplier', false);
    component.instance().handleInputChange('link', 'link.supplier.co.uk');
    component.instance().saveOrCreateSupplier();
    expect(saveSupplier.mock.calls).toHaveLength(1);
    expect(closeModal.mock.calls).toHaveLength(1);
    expect(saveSupplier.mock.calls[0][0].id).toBe(undefined);
    expect(saveSupplier.mock.calls[0][0].supplier_name).toBe('Replacement name');
    expect(saveSupplier.mock.calls[0][0].brand_names).toBe(undefined);
    expect(saveSupplier.mock.calls[0][0].link).toBe('link.supplier.co.uk');
    expect(saveSupplier.mock.calls[0][0].preferred_supplier).toBe(false);
  });
  it('delete for existing supplier calls passed method', () => {
    const deleteSupplier = jest.fn();
    const closeModal = jest.fn();
    const supplier = {
      id: 1,
      supplier_name: 'supplier name',
      brand_names: ['brands 1', 'brand 2', 'brand 3'],
      link: 'supplier.co.uk',
      preferred_supplier: false,
    };
    const component = shallow(
      <SupplierEdit supplier={supplier} deleteSupplier={deleteSupplier} closeModal={closeModal} />,
    );

    component.instance().handleInputChange('supplier_name', 'Replacement name');
    component.instance().handleInputChange('preferred_supplier', true);
    component.instance().handleInputChange('link', 'link.supplier.co.uk');
    component.instance().deleteOrRemoveSupplier();
    expect(deleteSupplier.mock.calls).toHaveLength(1);
    expect(closeModal.mock.calls).toHaveLength(1);
    expect(deleteSupplier.mock.calls[0][0]).toBe(supplier.id);
  });
  it('delete on modal for new supplier just closes modal', () => {
    const deleteSupplier = jest.fn();
    const closeModal = jest.fn();
    const supplier = {};
    const component = shallow(
      <SupplierEdit supplier={supplier} deleteSupplier={deleteSupplier} closeModal={closeModal} />,
    );

    component.instance().handleInputChange('supplier_name', 'Replacement name');
    component.instance().handleInputChange('preferred_supplier', true);
    component.instance().handleInputChange('link', 'link.supplier.co.uk');
    component.instance().deleteOrRemoveSupplier();
    expect(deleteSupplier.mock.calls).toHaveLength(0);
    expect(closeModal.mock.calls).toHaveLength(1);
  });
});
