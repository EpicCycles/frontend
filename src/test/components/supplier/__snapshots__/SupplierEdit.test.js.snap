// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierEdit displays correctly for existing supplier 1`] = `
<Fragment>
  <div
    style={
      Object {
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    <h2>
      Edit Supplier
    </h2>
    <EditModelPage
      model={
        Object {
          "brand_names": Array [
            "brands 1",
            "brand 2",
            "brand 3",
          ],
          "id": 1,
          "link": "supplier.co.uk",
          "preferred_supplier": false,
          "supplier_name": "supplier name",
        }
      }
      modelFields={
        Array [
          Object {
            "error": "A value is required.",
            "fieldName": "supplier_name",
            "header": "Supplier",
            "length": 100,
            "required": true,
            "size": 20,
            "type": "text",
          },
          Object {
            "fieldName": "preferred_supplier",
            "header": "Preferred",
            "type": "checkbox",
          },
          Object {
            "fieldName": "link",
            "header": "URL",
            "length": 100,
            "size": 40,
            "type": "text",
            "validator": [Function],
          },
        ]
      }
      onChange={[Function]}
      persistedModel={
        Object {
          "brand_names": Array [
            "brands 1",
            "brand 2",
            "brand 3",
          ],
          "id": 1,
          "link": "supplier.co.uk",
          "preferred_supplier": false,
          "supplier_name": "supplier name",
        }
      }
      showReadOnlyFields={true}
    />
    <div
      key="supplierBrands1"
    >
      Brands: 
    </div>
    <div
      className="align_right"
    >
      <ModelEditIcons
        componentKey={1}
        model={
          Object {
            "brand_names": Array [
              "brands 1",
              "brand 2",
              "brand 3",
            ],
            "id": 1,
            "link": "supplier.co.uk",
            "preferred_supplier": false,
            "supplier_name": "supplier name",
          }
        }
        modelDelete={[Function]}
        modelReset={[Function]}
        modelSave={[Function]}
      />
    </div>
  </div>
</Fragment>
`;

exports[`SupplierEdit displays correctly for existing supplier as a modal dialogue 1`] = `
<Fragment>
  <div
    style={
      Object {
        "textAlign": "right",
        "width": "100%",
      }
    }
  >
    <Icon
      as="i"
      circular={true}
      link={true}
      name="remove"
      onClick={[MockFunction]}
    />
  </div>
  <div
    style={
      Object {
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    <h2>
      Edit Supplier
    </h2>
    <EditModelPage
      model={
        Object {
          "brand_names": Array [
            "brands 1",
            "brand 2",
            "brand 3",
          ],
          "id": 1,
          "link": "supplier.co.uk",
          "preferred_supplier": false,
          "supplier_name": "supplier name",
        }
      }
      modelFields={
        Array [
          Object {
            "error": "A value is required.",
            "fieldName": "supplier_name",
            "header": "Supplier",
            "length": 100,
            "required": true,
            "size": 20,
            "type": "text",
          },
          Object {
            "fieldName": "preferred_supplier",
            "header": "Preferred",
            "type": "checkbox",
          },
          Object {
            "fieldName": "link",
            "header": "URL",
            "length": 100,
            "size": 40,
            "type": "text",
            "validator": [Function],
          },
        ]
      }
      onChange={[Function]}
      persistedModel={
        Object {
          "brand_names": Array [
            "brands 1",
            "brand 2",
            "brand 3",
          ],
          "id": 1,
          "link": "supplier.co.uk",
          "preferred_supplier": false,
          "supplier_name": "supplier name",
        }
      }
      showReadOnlyFields={true}
    />
    <div
      key="supplierBrands1"
    >
      Brands: 
    </div>
    <div
      className="align_right"
    >
      <ModelEditIcons
        componentKey={1}
        model={
          Object {
            "brand_names": Array [
              "brands 1",
              "brand 2",
              "brand 3",
            ],
            "id": 1,
            "link": "supplier.co.uk",
            "preferred_supplier": false,
            "supplier_name": "supplier name",
          }
        }
        modelDelete={[Function]}
        modelReset={[Function]}
        modelSave={[Function]}
      />
    </div>
  </div>
</Fragment>
`;

exports[`SupplierEdit displays correctly for new supplier 1`] = `
<Fragment>
  <div
    style={
      Object {
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    <h2>
      Edit Supplier
    </h2>
    <EditModelPage
      model={Object {}}
      modelFields={
        Array [
          Object {
            "error": "A value is required.",
            "fieldName": "supplier_name",
            "header": "Supplier",
            "length": 100,
            "required": true,
            "size": 20,
            "type": "text",
          },
          Object {
            "fieldName": "preferred_supplier",
            "header": "Preferred",
            "type": "checkbox",
          },
          Object {
            "fieldName": "link",
            "header": "URL",
            "length": 100,
            "size": 40,
            "type": "text",
            "validator": [Function],
          },
        ]
      }
      onChange={[Function]}
      persistedModel={Object {}}
      showReadOnlyFields={true}
    />
    <div
      key="supplierBrands"
    >
      Brands: 
    </div>
    <div
      className="align_right"
    >
      <ModelEditIcons
        componentKey=""
        model={Object {}}
        modelDelete={[Function]}
        modelReset={[Function]}
        modelSave={[Function]}
      />
    </div>
  </div>
</Fragment>
`;
